### ISSUE TITLE
Mention full type in diagnostic on `call-top-callable`

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3059]
### Summary

I've noticed on a couple of narrowing-related issues (#2963, #3056) that the current message for `call-top-callable` is rather unhelpful when the called type is a narrowed/more complex one:
https://play.ty.dev/d7981a15-d30c-4a37-8429-643fdf0b3731
```py
from typing import Callable
def f(x: int | Callable[[], int]):
    if callable(x):
        reveal_type(x)  # Revealed type: `(int & Top[(...) -> object]) | (() -> int)`
        x()  # Object of type `Top[(...) -> object]` is not safe to call; its signature is not known (call-top-callable) 
```
The error on `x()` doesn't mention the actual type of `x` at all, which makes it easy to assume ty is failing to narrow the type completely, instead of it being a portion of the overall type that is causing a problem. If the type being called is more complex than just `Top[(...) -> something]`, I think an error message like this would be better:
```
Object of type `<type of the object>` is not safe to call due to `<existing diagnostic type>`; its signature is partially unknown (call-top-callable) 
```
So on the example above, that would now look like this:
```py
from typing import Callable
def f(x: int | Callable[[], int]):
    if callable(x):
        x()  # Object of type `(int & Top[(...) -> object]) | (() -> int)` is not safe to call due to `Top[(...) -> object]`; its signature is partially unknown (call-top-callable) 
```

### Version

playground e637bbcfc

**Comment by dhruvmanila** [comment]
Just for context, we do include the type at the info messages but that's only visible using the full diagnostic output. I agree that it would be useful to include the entire type followed by the specific non-callable type.

```py
error[call-top-callable]: Object of type `Top[(...) -> object]` is not safe to call; its signature is not known
 --> main.py:5:9
  |
3 |     if callable(x):
4 |         reveal_type(x)  # Revealed type: `(int & Top[(...) -> object]) | (() -> int)`
5 |         x()  # Object of type `Top[(...) -> object]` is not safe to call; its signature is not known (call-top-callable) 
  |         ^^^
  |
info: This type includes all possible callables, so it cannot safely be called because there is no valid set of arguments for it
info: Intersection element `Top[(...) -> object]` is incompatible with this call site
info: Attempted to call intersection type `int & Top[(...) -> object]`
info: Attempted to call union type `(int & Top[(...) -> object]) | (() -> int)`
info: rule `call-top-callable` is enabled by default
```
